src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/security/JWTHeaderClient.kt

```diff
@@ -0,0 +1,23 @@
+package org.vaccineimpact.orderlyweb.security
+
+import org.pac4j.core.credentials.extractor.HeaderExtractor
+import org.pac4j.http.client.direct.HeaderClient
+import org.vaccineimpact.orderlyweb.models.ErrorInfo
+
+class JWTHeaderClientWrapper(helper: TokenVerifier) : OrderlyWebCredentialClientWrapper
+{
+    override val errorInfo = ErrorInfo("bearer-token-invalid", "Bearer token not supplied in Authorization header, or bearer token was invalid")
+    override val client = JWTHeaderClient(helper)
+}
+
+// This client receives the token as TokenCredentials and stores the result as JwtProfile
+class JWTHeaderClient(helper: TokenVerifier) : HeaderClient(
+        "Authorization",
+        "Bearer ",
+        OrderlyWebBearerTokenAuthenticator(helper.signatureConfiguration, helper.expectedIssuer))
+{
+    init
+    {
+        credentialsExtractor = HeaderExtractor(headerName, prefixHeader, name)
```

Comment:
Even simpler, this whole `init` block is now obsolete, can totally delete the body of this class!